src/thermostate/vapordome.py

```diff
@@ -0,0 +1,245 @@
+get_ipython().run_line_magic('matplotlib', 'notebook')
+from thermostate import State, units
+from CoolProp.CoolProp import PropsSI
+import matplotlib.pyplot as plt
+import matplotlib.lines as lines
+import numpy as np
+
+
+from dataclasses import dataclass, field
+
+@dataclass
+class PlottedState:
+    # https://realpython.com/python-data-classes/
+    key: str
+    state: State
+    # key: Plot axes string (Tv, pv)
+    # value: Line2D instance for that plot of the marker for this state
+    markers: dict = field(default_factory=dict)
+
+class VaporDome:
+    axis_units = {"v": "m**3/kg", "T": "K", "s": "J/(kg*K)", "p": "pascal", "u": "J/kg","h": "J/kg","x": "dimensionless" }
+    def __init__(self, *args):
+        min_temperature = PropsSI('Tmin','water')
+        max_temperature = PropsSI('Tcrit','water')
+        T_range = np.logspace(np.log10(min_temperature), np.log10(max_temperature),400)*units.K
+        self.st_f = [State("water", T=T, x=0*units.dimensionless) for T in T_range]
+        self.st_g = [State("water", T=T, x=1*units.dimensionless) for T in T_range]
+        self.states = {}
+        self.plots = {}
+        self.processes = {}
+        for axes in args:
+            self.plot(axes[0],axes[1])
+    def plot(self, x_axis, y_axis):
+        if x_axis + y_axis not in self.plots:
+            fig, axis = plt.subplots()
+            self.plots[x_axis + y_axis] = (fig, axis)
+            
+            x_f = [getattr(st, x_axis).magnitude for st in self.st_f]
+            x_f = np.array(x_f)*getattr(units, self.axis_units[x_axis])
+            y_f = [getattr(st, y_axis).magnitude for st in self.st_f]
+            y_f = np.array(y_f)*getattr(units, self.axis_units[y_axis])
+            axis.plot(x_f, y_f)
+
+            x_g = np.array([getattr(st, x_axis).magnitude for st in self.st_g])*getattr(units, self.axis_units[x_axis])
+            y_g = np.array([getattr(st, y_axis).magnitude for st in self.st_g])*getattr(units, self.axis_units[y_axis])
+            axis.plot(x_g, y_g)
+            if x_axis == 'p' or x_axis == 'v':
```

Comment:
You can simplify this to say `if x_axis in ("p", "v"):`. Likewise for the `y_axis` below.